src/common/bitstream.cpp

```diff
@@ -0,0 +1,392 @@
+#include <algorithm>
+#include "bitstream.h"
+#include "util.h"
+
+#define BITS_TO_BYTES(bits) (((bits) + 7) / 8)
+
+
+BitStream::BitStream(std::vector<uint8_t>& exisitingBuf) :
+    buf(exisitingBuf),
+    streamBitPos(0),
+    lastError(Error::NONE) {
+
+}
+
+size_t BitStream::getSizeBits() const {
+    return buf.size() * 8;
+}
+
+size_t BitStream::getRemainingBits() const {
+    size_t sizeBits = getSizeBits();
+    if(streamBitPos >= sizeBits) {
+        return 0;
+    }
+    else {
```

Comment:
We should keep our else-newlining consistent, so one of us needs to change our settings. I like to keep them on the same line to prevent this strange kind of dangling closing brace.
Tools->Options->Text Editor->C/C++->Formatting->New Lines->Position of Keywords->uncheck Place 'else' on a new line in VC++ will make it stop shunting them to a new line.